rollout/controller.go

```diff
@@ -346,6 +347,81 @@ func (c *Controller) syncHandler(key string) error {
 		return nil
 	}
 
+	referencedResources := validation.ReferencedResources{}
+
+	if r.Spec.Strategy.BlueGreen != nil {
+		blueGreen := r.Spec.Strategy.BlueGreen
+		_, _, err := c.getPreviewAndActiveServices(r)
+		if err != nil {
+			return err
+		}
+		for _, template := range blueGreen.PrePromotionAnalysis.Templates {
+			analysisTemplate, err := c.argoprojclientset.ArgoprojV1alpha1().AnalysisTemplates(r.Namespace).Get(template.TemplateName, metav1.GetOptions{})
```

Comment:
I think we should use an analysistemplate lister instead of the API client. The former does not incur a API cost